### PR TITLE
Feature/sites 21278

### PR DESCRIPTION
**Pull request by alwinjoseph02**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: src/main/java/com/adobe/cq/commerce/graphql/client/GraphqlClient.java [hunk 1]
```diff
@@ -112,4 +112,14 @@ public interface GraphqlClient {
      */
     public <T, U> GraphqlResponse<T, U> execute(GraphqlRequest request, Type typeOfT, Type typeOfU, RequestOptions options);
 
+    /**
+     * Delete cache entries for in-memory cache based on the cache entries.
```
Comment (LSantha) [inline src/main/java/com/adobe/cq/commerce/graphql/client/GraphqlClient.java:116]: This is a bit confusing. Could you reformulate it a b it more clearly what this method is supposed to do?
The idea was that when all params are null, it clears everything, when some params are not null it clears the machine entries, right?

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            }
+        }
+    }
+
+    private void invalidateStoreView(String storeView) {
+        caches.forEach((cacheName, cache) -> {
+            cache.asMap().entrySet().stream()
+                .filter(entry -> checkIfStorePresent(storeView, entry.getKey()))
+                .forEach(entry -> {
+                    LOGGER.info("Invalidating key: {} in cache: {}", entry.getKey(), cacheName);
```
Comment (LSantha) [inline src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java:63]: Pls. use debug level login here.

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                });
+        });
+    }
+
+    private void invalidateAll() {
+        LOGGER.info("Invalidating all caches...");
+        caches.values().forEach(Cache::invalidateAll);
+    }
+
+    private void invalidateSpecificCaches(String[] cacheEntries) {
```
Comment (LSantha) [inline src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java:74]: Pls. rename cacheEntries param to cacheNames.

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        for (String cacheName : cacheEntries) {
+            Cache<CacheKey, GraphqlResponse<?, ?>> cache = caches.get(cacheName);
+            if (cache != null) {
+                LOGGER.info("Invalidating cache: {}", cacheName);
```
Comment (LSantha) [inline src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java:78]: Pls. use debug level login here.

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                    }
+                    GraphqlResponse<?, ?> value = entry.getValue();
+                    String jsonResponse = gson.toJson(value);
+                    // Replace \\u003d with = in the JSON string
+                    jsonResponse = jsonResponse.replace("\\u003d", "=");
+                    Matcher matcher = regex.matcher(jsonResponse);
+                    return matcher.find();
+                })
+                .forEach(entry -> {
+                    LOGGER.info("Invalidating key: {} in cache: {}", entry.getKey(), cacheName);
```
Comment (LSantha) [inline src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java:106]: Pls. use debug level login here.

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java (hunk 2 above)
Comment (alwinjoseph02) [reply src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java]: Done

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java (hunk 3 above)
Comment (alwinjoseph02) [reply src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java]: Done

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java (hunk 4 above)
Comment (alwinjoseph02) [reply src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java]: Done

File: src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java (hunk 5 above)
Comment (alwinjoseph02) [reply src/main/java/com/adobe/cq/commerce/graphql/client/impl/CacheInvalidator.java]: Done

File: src/main/java/com/adobe/cq/commerce/graphql/client/GraphqlClient.java (hunk 1 above)
Comment (alwinjoseph02) [reply src/main/java/com/adobe/cq/commerce/graphql/client/GraphqlClient.java]: Done

**Review by LSantha — APPROVED** [review]

**Pull request merged**